File: src/main/java/com/github/sorusclient/client/module/impl/perspective/v1_8_9/PerspectiveHook.java
package com.github.sorusclient.client.module.impl.perspective.v1_8_9;

import com.github.sorusclient.client.Sorus;
import com.github.sorusclient.client.module.ModuleManager;
import com.github.sorusclient.client.module.impl.perspective.Perspective;
import v1_8_9.net.minecraft.util.math.MathHelper;

public class PerspectiveHook {

    private static boolean x = true;

    public static float yaw = 0;
    public static float prevYaw = 0;
    public static float pitch = 0;
    public static float prevPitch = 0;

    public static float modifyDelta(float delta) {
        Perspective perspective = Sorus.getInstance().get(ModuleManager.class).get(Perspective.class);

        boolean isYaw = x;
        x = !x;

        if (perspective.isEnabled() && perspective.isToggled()) {
            if (isYaw) {
                float oldYaw = yaw;
                yaw = (float)((double)yaw + (double) delta * 0.15D);
                prevYaw += yaw - oldYaw;
            } else {
                float oldPitch = pitch;
                pitch = (float)((double)pitch - (double) delta * 0.15D);
                pitch = MathHelper.clamp(pitch, -90.0F, 90.0F);
                prevPitch += pitch - oldPitch;
            }

            return 0;
        } else {
            return delta;
        }
    }

    public static float modifyPitch(float pitch) {
        Perspective perspective = Sorus.getInstance().get(ModuleManager.class).get(Perspective.class);

        if (perspective.isEnabled() && perspective.isToggled()) {
            return PerspectiveHook.pitch;
        }
        return pitch;
    }

    public static float modifyYaw(float yaw) {
        Perspective perspective = Sorus.getInstance().get(ModuleManager.class).get(Perspective.class);

        if (perspective.isEnabled() && perspective.isToggled()) {
            return PerspectiveHook.yaw;
        }
        return yaw;
    }

    public static float modifyPrevPitch(float prevPitch) {
        Perspective perspective = Sorus.getInstance().get(ModuleManager.class).get(Perspective.class);

        if (perspective.isEnabled() && perspective.isToggled()) {
            return PerspectiveHook.prevPitch;
        }
        return prevPitch;
    }

    public static float modifyPrevYaw(float prevYaw) {
        Perspective perspective = Sorus.getInstance().get(ModuleManager.class).get(Perspective.class);

        if (perspective.isEnabled() && perspective.isToggled()) {
            return PerspectiveHook.prevYaw;
        }
        return prevYaw;
    }

}

File: version/1.8.9/src/main/kotlin/com/github/sorusclient/client/feature/impl/perspective/v1_8_9/PerspectiveHook.kt
package com.github.sorusclient.client.feature.impl.perspective.v1_8_9

import com.github.sorusclient.client.feature.impl.perspective.Perspective
import v1_8_9.net.minecraft.util.math.MathHelper

@Suppress("UNUSED")
object PerspectiveHook {

    private var x = true
    var yaw = 0f
    var prevYaw = 0f
    var pitch = 0f
    var prevPitch = 0f

    @JvmStatic
    fun modifyDelta(delta: Float): Float {
        val isYaw = x
        x = !x
        return if (Perspective.isEnabled() && Perspective.isToggled) {
            if (isYaw) {
                val oldYaw = yaw
                yaw = (yaw.toDouble() + delta.toDouble() * 0.15).toFloat()
                prevYaw += yaw - oldYaw
            } else {
                val oldPitch = pitch
                pitch = (pitch.toDouble() - delta.toDouble() * 0.15).toFloat()
                pitch = MathHelper.clamp(pitch, -90.0f, 90.0f)
                prevPitch += pitch - oldPitch
            }
            0F
        } else {
            delta
        }
    }

    @JvmStatic
    fun modifyPitch(pitch: Float): Float {
        return if (Perspective.isEnabled() && Perspective.isToggled) {
            PerspectiveHook.pitch
        } else pitch
    }

    @JvmStatic
    fun modifyYaw(yaw: Float): Float {
        return if (Perspective.isEnabled() && Perspective.isToggled) {
            PerspectiveHook.yaw
        } else yaw
    }

    @JvmStatic
    fun modifyPrevPitch(prevPitch: Float): Float {
        return if (Perspective.isEnabled() && Perspective.isToggled) {
            PerspectiveHook.prevPitch
        } else prevPitch
    }

    @JvmStatic
    fun modifyPrevYaw(prevYaw: Float): Float {
        return if (Perspective.isEnabled() && Perspective.isToggled) {
            PerspectiveHook.prevYaw
        } else prevYaw
    }

}